bn_testing/dags.py

```diff
@@ -38,6 +39,8 @@ def __init__(self, n_nodes, n_groups=1, p=0.01, random_state=10):
         self.groups = {
             group_name: self._generate_node_names(
                 # TODO: Improve splitting
+                # How about specifying nodes_per_group and avoid
+                # that size of the last group is uncontrolled
```

Comment:
How about specifying nodes_per_group and avoid that size of the last group is uncontrolled ?

Reply:
thats a good idea. It may also make sense to allow groups of different sizes.

Reply:
maybe the user can specify a function that specifies the distribution for ex `lambda x: 5`